src/audio/source.rs

```diff
@@ -0,0 +1,84 @@
+//! Provides structures used to load audio files.
+
+use assets::*;
+use std::sync::Arc;
+
+/// A loaded audio file
+pub struct Source {
+    pub(crate) reference: AudioDataReference
+}
+
+/// This structure wraps the Arc Vec allowing us to provide a AsRef<[u8]> implementation.
+/// This was necessary in order to allow the rodio decoder to access the data.
+#[derive(Clone)]
+pub(crate) struct AudioDataReference {
+    pub raw: Arc<Vec<u8>>
+}
+
+impl AsRef<[u8]> for AudioDataReference {
+    fn as_ref(&self) -> &[u8] {
+        &*self.raw
+    }
+}
+
+impl Asset for Source {
+    type Context = ();
+    type Data = Vec<u8>;
+    type Error = NoError;
+
+    fn category() -> &'static str {
+        "audio"
+    }
+
+    fn from_data(data: Vec<u8>, _: &()) -> Result<Self, Self::Error> {
+        Ok(Source { reference: AudioDataReference { raw: Arc::new(data) } })
+    }
+}
+
+/// Loads audio from wav files.
+pub struct WavFormat;
```

Comment:
But I'm also not sure whether or not formats should belong to `amethyst_assets`.